### ISSUE TITLE
missing-home-modules warning with hs-boot files

### ISSUE THREAD

**Issue by thufschmitt** [tweag/rules_haskell#1164]
**Describe the bug**

Using a `.hs-boot` file in a `haskell_library` rule causes GHC to raise a warning:

```
<no location info>: warning: [-Wmissing-home-modules]
    Modules are not listed in command line but needed for compilation:
        A
```

**To Reproduce**

```
bazel build //tests/hs-boot:hs-boot-lib
```

**Expected behavior**

The build shouldn't cause any warning to be raised

**Environment**
 - OS name + version: NixOs 19.09
 - Bazel version: 1.1.0
 - Version of the rules: master

**Additional context**

Since the `hs-boot` files aren't explicitely passed to GHC, my first attempt at fixing this was to simply add them to the argument list, but this results in an error: 

```
    module ‘testsZShs-bootZShs-boot-lib:A’ is defined in multiple files: bazel-out/k8-fastbuild/bin/tests/hs-boot/srcs/A.hs
                                                                         bazel-out/k8-fastbuild/bin/tests/hs-boot/srcs/A.hs-boot
```

So I guess there's something more to it

**Comment by mboes** [comment]
This is an upstream issue: https://gitlab.haskell.org/ghc/ghc/issues/16551.

**Comment by aherrmann** [comment]
This seems to have been fixed upstream since GHC 8.10.1: https://gitlab.haskell.org/ghc/ghc/-/commit/c8380a4a738e5c2488337496b0d1b1faf6a7de9d
